commit 439bc8d0af866af6dd0face356bf6074c9ac51f1
Author: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>
Date:   Fri Jul 24 10:44:05 2020 +0200

    cxgb3: fix declaration of free_context
    
    free_context got moved to another struct by
    a4c102dd6b05 (verbs: Move free_context from verbs_device_ops to verbs_context_ops)
    
    Backport the change to older providers we maintain
    
    Signed-off-by: Nicolas Morey-Chaisemartin <nmoreychaisemartin@suse.com>

diff --git providers/cxgb3/iwch.c providers/cxgb3/iwch.c
index 6f3c8b9f1439..2bfca4f58f21 100644
--- providers/cxgb3/iwch.c
+++ providers/cxgb3/iwch.c
@@ -75,6 +75,8 @@ static const struct verbs_match_ent hca_table[] = {
 	{},
 };
 
+static void iwch_free_context(struct ibv_context *ibctx);
+
 static const struct verbs_context_ops iwch_ctx_common_ops = {
 	.query_device = iwch_query_device,
 	.query_port = iwch_query_port,
@@ -98,6 +100,7 @@ static const struct verbs_context_ops iwch_ctx_common_ops = {
 	.detach_mcast = iwch_detach_mcast,
 	.post_srq_recv = iwch_post_srq_recv,
 	.req_notify_cq = iwch_arm_cq,
+	.free_context = iwch_free_context,
 };
 
 static const struct verbs_context_ops iwch_ctx_t3a_ops = {
@@ -264,6 +267,5 @@ static const struct verbs_device_ops iwch_dev_ops = {
 	.alloc_device = iwch_device_alloc,
 	.uninit_device = iwch_uninit_device,
 	.alloc_context = iwch_alloc_context,
-	.free_context = iwch_free_context,
 };
 PROVIDER_DRIVER(cxgb3, iwch_dev_ops);
